Dockerfile

```diff
@@ -0,0 +1,32 @@
+FROM python:3.11.5-alpine as base
+
+RUN mkdir /usr/src/app
+WORKDIR /usr/src/app
+
+EXPOSE 5000
+
+RUN apk add curl
+RUN curl -sSL https://install.python-poetry.org | python3 -
+
+ENV PATH $PATH:/root/.local/bin
+
+FROM base as production
+
+COPY todo_app ./todo_app
+COPY poetry.lock pyproject.toml ./
+
+RUN poetry install
```

Comment:
I need to run `poetry install` for all of them, however, for the prod one I can't do it until I have the files copied. Is there a nicer way to go about this?

Reply:
`poetry install` does rely on your dependencies being documented, but not everything. So we can copy in the `pyproject.toml` and `poetry.lock` files before installing, and then copying the rest of the files later. I'd also probably handle that in the base stage, since I'm a fan of installing the dependencies in the build rather than the entrypoint*
```Dockerfile
ENV PATH $PATH:/root/.local/bin
COPY poetry.lock pyproject.toml ./
RUN poetry install

FROM base as production

COPY todo_app ./todo_app
```

*That said, this does mean you need to rebuild the container if you add a new library, so it's arguably more painful if that's a regular thing you're doing in dev, so definitely there's a trade off

Reply:
Bonus to note if we take this approach - because we install before the files are present, Python builds it's cache slightly differently and doesn't immediately recognise the created files as modules. Because of that, I think we should add empty `__init__.py` files to your various test directories so that it acknowledges those even if they weren't present before `poetry install` was run